lib/iris/tests/stock/_stock_2d_latlons.py

```diff
@@ -341,10 +341,28 @@ def adjust_coord(coord):
 
     adjust_coord(x_coord)
     adjust_coord(y_coord)
-    # Also mask the relevant data point.
-    data = cube.data  # N.B. fetch all the data.
-    if not ma.isMaskedArray(data):
-        # Promote to masked array, to avoid converting mask to NaN.
-        data = ma.masked_array(data)
-    data[at_iy, at_ix] = ma.masked
-    cube.data = data
+
+    # Check which dimensions are spanned by each coordinate.
+    for coord in (x_coord, y_coord):
+        if isinstance(coord, int):
```

Comment:
The 'coord is an int' option is _definitely_ not applicable here, as x_coord and y_coord are definitely coords in this usage.
I think the use of 'set' is also redundant.
I think it could just have ...
```
coord_dims = cube.coord_dims(coord)
if not coord_dims:
    ...
sliced_cube = cube.slices(coord_dims)
```
(needs checking though ?)

Reply:
I think you're probably right.  I remember having that conversation.